src/internal/_xtake.js

```diff
@@ -15,7 +15,8 @@ module.exports = (function() {
       return _reduced(result);
     } else {
       this.n -= 1;
-      return this.xf['@@transducer/step'](result, input);
+      return this.n === 0 ? _reduced(this.xf['@@transducer/step'](result, input))
+                          : this.xf['@@transducer/step'](result, input);
     }
```

Comment:
Ya, it is quite a bit going on between dispatching for Fantasy Land and the same functions supporting transducers vs their normal behavior, I get lost pretty much every time I come back to it :p

This line `this.xf['@@transducer/step'](result, input))` basically calls the next function in the transducer stack. If we return without calling that, we end up back [in one of our reducing functions (which one depending on the input)](https://github.com/ramda/ramda/blob/master/src/internal/_reduce.js). In that file, you will also see the check for the special `reduced` value, which breaks out of the reducing.